scylla/src/transport/session.rs

```diff
@@ -65,6 +65,62 @@ pub struct Session {
     auto_await_schema_agreement_timeout: Option<Duration>,
 }
 
+impl std::fmt::Debug for Session {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        struct ClusterNeatDebug<'a>(&'a Cluster);
+        impl<'a> std::fmt::Debug for ClusterNeatDebug<'a> {
+            fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                struct ClusterDataNeatDebug(Arc<ClusterData>);
+                impl std::fmt::Debug for ClusterDataNeatDebug {
+                    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                        let cluster_data = &self.0;
```

Comment:
Why not make these the default `Debug` implementations for `Cluster` and `ClusterData`?
Nesting `Debug` impls here makes the code hard to read, and they look pretty reusable, so why hide them here.

Reply:
if you make it the default, it would be impossible to print very detailed debug information from cluster and cluster data - the idea was to print a summary here, when printing Session, but if somebody wants to specifically print Cluster or ClusterData debug information, they should get non-truncated output. I'm fine with making ClusterDataNeatDebug and ClusterNeatDebug public though, as helpers.